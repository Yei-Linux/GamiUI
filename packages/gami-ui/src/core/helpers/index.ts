export * from './date-formatter'
export * from './validations.helper'
