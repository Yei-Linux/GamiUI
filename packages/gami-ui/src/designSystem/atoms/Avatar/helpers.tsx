import { TextModeType } from 'core/domain/types'
import { avatarTextModes } from './constants'
import { REGEX_RULES } from 'core/utils/constants'

export const validateMaxLetters = (
  firstLettersGot: string,
  maxSizeLetters: number
) => {
  const size = firstLettersGot.length

  if (size > maxSizeLetters) {
    return firstLettersGot.substring(0, maxSizeLetters)
  }

  return firstLettersGot
}

export const showOnlyFirstLettersOnText = (
  text: string,
  type: TextModeType,
  maxSizeLetters: number
): string => {
  const regexType = avatarTextModes[type]
  const firstLettersCaptured: string[] | null = text.match(
    REGEX_RULES[regexType]
  )

  const joinFirstLetters = firstLettersCaptured?.join('') || ''

  return validateMaxLetters(joinFirstLetters, maxSizeLetters)
}
