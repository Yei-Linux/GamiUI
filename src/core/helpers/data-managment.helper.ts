export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
