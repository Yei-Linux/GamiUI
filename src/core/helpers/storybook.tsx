import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};