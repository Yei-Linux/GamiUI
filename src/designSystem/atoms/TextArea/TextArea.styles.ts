import styled from "@emotion/styled";
import {
  inputBasicStyle,
  inputGeneralStyle,
} from "../../../styles/utilities/input";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const TextAreaWrapper = styled.textarea<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  margin: 0px;
  height: 69px;
  min-height: 75px;
  max-height: 225px;
  width: 100%;
  min-width: 100%;
  max-width: 100%;
  padding-top: 1rem;

  ${inputBasicStyle()};
  ${inputGeneralStyle()};
  ${(props: any) => twinStyles(props)};
`;
