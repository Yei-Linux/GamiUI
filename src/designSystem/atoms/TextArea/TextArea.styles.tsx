import styled from '@emotion/styled'
import {
  BorderType,
  FontWeightType,
  HeightType,
  ShadowType,
  TextAlignType,
  WidthType,
} from 'core/domain/types'
import { mixinInput } from 'styles/mixins/input'
import { spacing } from 'styles/tokens'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'

export const TextArea = styled.textarea<{
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
  $textAlign?: TextAlignType
  $fontWeight?: FontWeightType
}>`
  height: 69px;
  min-height: 75px;
  max-height: 225px;

  margin: ${spacing.margin.none};
  padding-top: ${spacing.padding.md};

  ${mixinInput('light')};

  ${({ $border, $shadow, $fontWeight, $width, $textAlign, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'lg',
      shadow: $shadow || 'sm',
      width: $width || 'auto',
      heigth: $heigth || 'sm',
      fontWeight: $fontWeight,
      textAlign: $textAlign,
    })};
`
