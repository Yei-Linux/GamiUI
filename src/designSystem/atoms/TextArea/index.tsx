export { default } from "./TextArea";
export { options } from "./constants";
