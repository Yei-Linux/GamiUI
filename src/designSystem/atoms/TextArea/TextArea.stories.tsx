import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import TextArea from ".";

const Template = getTemplate(TextArea);

export default {
  title: "Atoms/TextArea",
  component: TextArea,
  args: { ...TextArea.defaultProps },
  argTypes: {
    placeholder: { control: "text" },
    ...stylesControl,
  },
} as Meta;

export const BasicTextArea = Template.bind({});
BasicTextArea.args = {
  placeholder: "Type your text",
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM"
};