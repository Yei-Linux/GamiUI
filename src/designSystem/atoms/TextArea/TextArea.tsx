import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { TextAreaWrapper } from "./TextArea.styles";

interface AutoSize {
  minRows: number;
  maxRows: number;
}

export interface ITextArea extends IGeneralProps {
  /**
   * Placeholder TextArea to show
   */
  placeholder?: string;
  /**
   * Identifier TextArea
   */
  name?: string;
  /**
   * Value TextArea
   */
  value?: any;
  /**
   * Cols TextArea
   */
  cols?: number;
  /**
   * Rows TextArea
   */
  rows?: number;
  /**
   * Autosize TextArea
   */
  autoSize?: AutoSize;
  /**
   * Function to detect changes
   */
  onChangeFormItem?: any;
}

const TextArea = ({ autoSize, onChangeFormItem, ...args }: ITextArea) => {
  return (
    <TextAreaWrapper
      {...args}
      onChange={(e: any) => {
        onChangeFormItem(e.target.value);
        e.preventDefault();
      }}
    />
  );
};

TextArea.defaultProps = {
  name: "gamification",
  placeholder: "",
  cols: 5,
  rows: 5,
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};

export default TextArea;
