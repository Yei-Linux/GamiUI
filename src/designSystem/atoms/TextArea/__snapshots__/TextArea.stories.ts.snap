// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/TextArea Basic Text Area 1`] = `
<textarea
  className="css-155tpye-TextAreaWrapper eieaaae0"
  cols={5}
  name="gamification"
  onChange={[Function]}
  placeholder="Type your text"
  rows={5}
  width="NORMAL"
/>
`;
